### PR TITLE
overrides: add poetry build system for gitlab-lint

### PR DESCRIPTION
**Pull request by arichtman**
[Contribution](README.md#contributing) checklist (recommended but not always applicable/required):

- [ ] There's an _[automated test](tests/default.nix)_ for this change
- [ ] Commit messages or code include _references to related issues or PRs_ (including third parties)
- [x] Commit messages are _[conventional](https://www.conventionalcommits.org/)_ - examples from [the log](https://github.com/nix-community/poetry2nix/commits/master) include "feat: add changelog files to fixup hook", "fix(contourpy): allow wheel usage", and "test: add sqlalchemy2 test"

### CODE REVIEW

File: overrides/build-systems.json [hunk 1]
```diff
@@ -7089,6 +7089,9 @@
   "gitlike-commands": [
     "poetry-core"
   ],
+  "gitlab-lint" : [
+    "poetry"
+  ],
```
Comment (fpletz) [inline overrides/build-systems.json:7094]: Formatting checks fails. This should be put before `"gitlike-commands"`.

Comment (arichtman) [reply overrides/build-systems.json:7091]: thank you! done and rebased onto default branch.

File: overrides/build-systems.json [hunk 2]
```diff
@@ -7825,6 +7825,9 @@
   "gitignore-parser": [
     "setuptools"
   ],
+  "gitlab-lint" : [
+    "poetry"
```
Comment (cpcloud) [inline overrides/build-systems.json:7829]:
```suggestion
    "poetry-core"
```

**Comment by cpcloud** [comment]
Passing locally, merging.

**Pull request merged**